keps/sig-auth/740-service-account-external-signing/README.md

```diff
@@ -1,113 +1,605 @@
+  - Components depending on the feature gate: kube-apiserver
+- [x] Other
+  - Describe the mechanism:
+    - To enable:
+      - kube-apiserver will need a restart with `--service-account-key-file`/`--service-account-signing-key-file` being un-set and `--service-account-signing-endpoint` being set.
+      - former value of `--service-account-key-file` (a path on the file system) will need to be supplied to ExternalJWTSigner in some fashion(this is up to the owners of the distribution).
+      - ExternalJWTSigner needs to combine the supported keys that are listed from external key management solutions and the keys that were on the cluster before enablement.
+    - To disable:
+      - kube-apiserver will need a restart with `--service-account-key-file`/`--service-account-signing-key-file` being set and `--service-account-signing-endpoint` being un-set.
+      - Externally supported keys will need to get stitched into the set at `--service-account-key-file` (unless distributions want to stop supporting those keys completely).
+  - Will enabling / disabling the feature require downtime of the control
+    plane?
+    - No
+  - Will enabling / disabling the feature require downtime or reprovisioning
+    of a node?
+    - No
+
+###### Does enabling the feature change any default behavior?
+
+Not from an end-user perspective. 
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+It will be possible but will need additional work from respective distributions. Thus, each distribution can decide for themselves.
+
+Can be achieved by satisfying the following:
+- kube-apiserver will need a restart with `--service-account-key-file`/`--service-account-signing-key-file` set and `--service-account-signing-endpoint` un-set.
+- Distributions will require importing the externally supported keys to the file system path as configured in `--service-account-key-file`.
+- If distributions do not intend to support tokens signed by external signers after the feature is disabled, then they can come up with a fair warning.
+
+###### What happens if we re-enable the feature if it was previously rolled back?
+
+experience would be the same as when enabling.
+
+###### Are there any tests for feature enablement/disablement?
+
+- Unit and E2E tests will be added.
+- The tests would **not** focus on continuing to support the same key set when enabling/disabling but would rather focus on viability of enabling/disabling.
+
+### Rollout, Upgrade and Rollback Planning
+
+Since it is an opt-in feature; upgrading existing clusters to versions with this feature shall be a no-op in context of the feature.
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+Since it is an opt-in feature, existing clusters shall remain unaffected. However, since we are touching paths for JWT signing and authentication, if a bug slips in, it can disrupt the respective operation.
```

Comment:
The idea of this question is to imagine situation where you are enabling the feature and turning it on (with the kube-apiserver flag). What can potentially go wrong in that case?

Reply:
For example the external jwt signer can't run, what will happen?

Reply:
(this needs to be addressed)

Reply:
Done